### PR TITLE
SCT-3084 fix broken search in data panel

### PR DESCRIPTION
**Pull request by eapearson**
# Fixes data panel search feature

The data panel's search feature was non-functional. This was simply caused by incorrect usage of `entries()` which caused the data object list filtering attempt to fail.

This was reported in a public ticket (https://kbase-jira.atlassian.net/browse/PUBLIC-1687) with development linked to a tracking ticket (https://kbase-jira.atlassian.net/browse/SCT-3084).

This PR also includes a single test added to ensure basic filtering is operational. (More tests could be added to cover data panel filtering and sorting.)

## Jira Ticket / Issue #
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions

The test for this fix is included in `test/unit/spec/narrative_core/kbaseNarrativeDataList-spec.js` located here https://github.com/kbase/narrative/pull/2371/files#diff-cb9d9f24ad0c6a467ed27c2706878f2be2642c542281180cd8cfb5886bf572b7R206. This will be run with the frontend unit tests, `make test-frontend-unit`.

- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [-] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [-] Any dependent changes have been merged and published in downstream modules
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [-] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook

# Updating Version and Release Notes (if applicable)

- [-] [Version has been bumped](https://semver.org/) for each release
- [x] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

**Comment by briehl** [comment]
Thanks. Pretty much duplicates this: https://github.com/kbase/narrative/pull/2320 which has the fix, but I never found time to get tests together for it.

File: kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeDataList.js [hunk 1]
```diff
@@ -2095,52 +2095,62 @@ define([
...
+                term = term.replace(/\|/g, '\\|').replace(/\\\\\|/g, '|');
+
+                // dots are common in names, so we escape them, but
                 // if a user writes '\\.' we assume they want the regex '.'
+                term = term.replace(/\./g, '\\.').replace(/\\\\\./g, '.');
 
                 const regex = new RegExp(term, 'i');
 
+                // TODO: n_filteredObjsRendered does not seem to be used anywhere, but needs
+                // further investigation.
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeDataList.js:2109]: It can probably be deleted.

**Review by briehl — APPROVED** [review]

**Comment by briehl** [comment]
Any idea what's up with the public search tests failing? Is CI search down?

**Comment by eapearson** [comment]
> Any idea what's up with the public search tests failing? Is CI search down?

Yes, CI search had been down since yesterday evening, Keith brought it back up this morning.

**Pull request merged**